### PR TITLE
Feature/resource/okta group membership

### PR DESCRIPTION
**Pull request by jralmaraz**
Initial group membership resource.

### CODE REVIEW

**Comment by alkar** [comment]
Thanks for picking this up!

I wonder if we should instead have `user_ids` (a list of strings) to define group membership. This would result in less defined resources (one resource for multiple users in a group).

Additionally, it could also allow for managing admin group members exclusively, guaranteeing that only the defined users are members of a group. Perhaps this would make more sense being implemented as part of `okta_group`, similar to how security groups and security group rules are implemented in the aws provider?

**Comment by quantumew** [comment]
You can currently manage a list of users with `okta_group` using the `users` attribute. Additionally I think having a one to one resource should be the first order of business since it can be leveraged to service both needs. I do also think it is important to have yet another resource that can attach multiple users to an existing group. I think it is less pressing than the one to one resource.

**Comment by alkar** [comment]
> You can currently manage a list of users with `okta_group` using the `users` attribute. Additionally I think having a one to one resource should be the first order of business since it can be leveraged to service both needs. I do also think it is important to have yet another resource that can attach multiple users to an existing group. I think it is less pressing than the one to one resource.

Ah! I didn't check the source, only looked at the docs which are not updated!
After a bit more digging, I think this resource makes sense as-is since the API behaves in the same way ([add one user to one group](https://developer.okta.com/docs/reference/api/groups/#add-user-to-group)).

File: okta/resource_okta_group_membership.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		Delete: resourceGroupMembershipDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"user_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Use to associate group with",
```
Comment (rtlong) [inline okta/resource_okta_group_membership.go]: typo: `User`

Comment (jralmaraz) [reply okta/resource_okta_group_membership.go]: Thanks! Fixed on 76bca02.

**Comment by ocxo** [comment]
Thanks for the contribution @jralmaraz! This repo has officially moved to Hashicorp's collection of Terraform Providers at https://github.com/terraform-providers/terraform-provider-okta/. If you still wish to see this PR land on the provider, please re-open at https://github.com/terraform-providers/terraform-provider-okta/pulls.

**Pull request closed without merging**